app/src/main/java/com/example/android/codelabs/paging/data/GithubRepository.kt

```diff
@@ -17,60 +17,75 @@
 package com.example.android.codelabs.paging.data
 
 import android.util.Log
-import androidx.lifecycle.MutableLiveData
 import com.example.android.codelabs.paging.api.GithubService
 import com.example.android.codelabs.paging.api.searchRepos
-import com.example.android.codelabs.paging.db.GithubLocalCache
+import com.example.android.codelabs.paging.model.Repo
 import com.example.android.codelabs.paging.model.RepoSearchResult
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
 
 /**
  * Repository class that works with local and remote data sources.
  */
-class GithubRepository(
-    private val service: GithubService,
-    private val cache: GithubLocalCache
-) {
+@FlowPreview
+@ExperimentalCoroutinesApi
+class GithubRepository(private val service: GithubService) {
+
+    // keep the list of responses
+    private val inMemoryCache = ConflatedBroadcastChannel<RepoSearchResult>()
 
     // keep the last requested page. When the request is successful, increment the page number.
     private var lastRequestedPage = 1
 
-    // LiveData of network errors.
-    private val networkErrors = MutableLiveData<String>()
-
     // avoid triggering multiple requests in the same time
     private var isRequestInProgress = false
 
     /**
      * Search repositories whose names match the query.
      */
-    fun search(query: String): RepoSearchResult {
+    suspend fun search(query: String): Flow<RepoSearchResult> {
```

Comment:
This method is a bit confusing for me as the name is `search` (i.e. perform a search operation) but also returns a `inMemoryCache` as Flow with which you will receive the searched item + other updates that can happen.

Is that intended? Something is wrong with that for me, I think it's the name of the function, that should say more